File: frontend/app/vo/agents/AgentVo.ts
/**
 * 백엔드 com.jsh.aidesk.serverapi.agents.vo 와 1:1 대응되는 타입.
 * 응답 envelope (ResponseJson<T>) 의 data 필드 타입.
 */

export type AgentStatus = 'active' | 'idle' | 'waiting' | 'error';

/** channel/channel_backend.md §4 의 agent type 분류. */
export type AgentType = 'self' | 'me' | 'internal' | 'human' | 'colleague';

export interface AgentItem {
  agentId: string;
  agentName: string;
  workspaceDir: string;
  tmuxSession: string;
  status: AgentStatus;
  taskDesc: string | null;
  model: string;
  contextPct: number | null;
  startedAt: string;          // ISO 8601
  updatedAt: string | null;
  /** 소유 사용자 (멀티유저). 자체 채널 모델 도입 후 응답에 동봉. */
  ownerAccountSn?: number | null;
  /** 발신자의 시점에서 본 type. caller 정보 없으면 backend 가 기본 분류만 부여. */
  type?: AgentType | null;
}

export interface AgentSummary {
  total: number;
  active: number;
  waiting: number;
  idle: number;
  error: number;
}

export interface AgentListResponse {
  list: AgentItem[];
  summary: AgentSummary;
}

export interface AgentCreateRequest {
  agentName: string;
  workspaceDir: string;
  model: 'claude' | 'codex' | 'hermes';
}

/**
 * 공통 응답 envelope.
 */
export interface ApiEnvelope<T> {
  result: number;
  message: string;
  data: T;
  timestamp: string;
}

File: backend/src/main/java/com/jsh/aidesk/serverapi/agents/vo/AgentVo.java
package com.jsh.aidesk.serverapi.agents.vo;

import java.time.OffsetDateTime;

import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

@Getter
@Setter
@ToString
public class AgentVo {

    private String agentId;
    /** 에이전트 소유 사용자 (t_user.account_sn). multi-user 격리에서 모든 조회/생성이 이 키로 필터링. */
    private Long ownerAccountSn;
    private String agentName;
    private String workspaceDir;
    private String tmuxSession;
    private String status;
    private String taskDesc;
    private String model;
    private Integer contextPct;
    private boolean bootstrapApplied;
    private OffsetDateTime startedAt;
    private OffsetDateTime updatedAt;
    private OffsetDateTime deletedAt;
    /** 에이전트 분류 — internal / external / me / human. Phase 2 의 인증 분기 + UI 아이콘 기반. */
    private String agentType;
    /** 외부 AI 의 Bearer token BCrypt 해시. 인증 분기에서만 사용. response 직렬화는 절대 X. */
    private String bearerTokenHash;
    /** 외부 AI 의 현재 활성 Bearer token 발급 시각 (감사 + UI 표시). */
    private OffsetDateTime bearerTokenCreatedAt;
}
